Sources/ApolloCodegenLib/ASTOutput.swift

```diff
@@ -8,6 +8,12 @@ struct ASTOutput: Codable, Equatable {
   /// An array of all fragments to generate code for.
   let fragments: [ASTFragment]
   
-  /// An array of "all" types used <-- TODO: Figure out why some are not in here
```

Comment:
Sorry the ignorance, but have you figured it out why some of the types weren't there before?

Reply:
Yes, excellent question! 

The answer is that `typesUsed` was designed to be only for _Input_ types, though it was not really documented as such. Unions and interfaces were not included, and I couldn't figure out why when I initially started working through the JSON, which is why I left myself this `TODO`. 

In an upcoming version of code generation, union and interface types will be included - this PR works off some temporary JSON to support generating code for that.